src/Core/UserRuleSettings/AnalysisSettings.cs

```diff
@@ -57,14 +58,15 @@ public class AnalysisSettings
 {
     private const string AnyDirectoryWildcard = "**";
     private static readonly string AnyRootPrefix = AnyDirectoryWildcard + Path.AltDirectorySeparatorChar;
+    private readonly ImmutableList<string> userDefinedFileExclusions = ImmutableList<string>.Empty;
 
-    private readonly List<string> userDefinedFileExclusions = [];
-    [JsonProperty("sonarlint.rules", ObjectCreationHandling = ObjectCreationHandling.Reuse)]
-    public Dictionary<string, RuleConfig> Rules { get; set; } = new(StringComparer.OrdinalIgnoreCase);
+    [JsonProperty("sonarlint.rules")]
+    [JsonConverter(typeof(ImmutableDictionaryIgnoreCaseConverter<string, RuleConfig>))]
+    public ImmutableDictionary<string, RuleConfig> Rules { get; init; }
```

Comment:
Using immutables is better, so good job for doing that, but technically we could have gotten away with IReadOnlyBlaBla interfaces and copying, which would be less code changes